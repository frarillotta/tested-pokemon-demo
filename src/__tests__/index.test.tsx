import { render } from "@testing-library/react";
import { QueryClient, QueryClientProvider } from "react-query";
import Homepage from "../pages";

describe("Homepage", () => {
	
	it("should have a consistent snapshot", () => {
		const queryClient = new QueryClient();
		const pokemonsList = {
			"count":1118,
			"next":null,
			"previous":null,
			"results":[
				{"name":"bulbasaur","url":"https://pokeapi.co/api/v2/pokemon/1/"},
				{"name":"ivysaur","url":"https://pokeapi.co/api/v2/pokemon/2/"},
				{"name":"venusaur","url":"https://pokeapi.co/api/v2/pokemon/3/"}
			]
		};

		const {container} = render(<QueryClientProvider client={queryClient}>
			<Homepage pokemonsList={pokemonsList}/>
		</QueryClientProvider>)
		expect(container).toMatchSnapshot();
	})
	
})