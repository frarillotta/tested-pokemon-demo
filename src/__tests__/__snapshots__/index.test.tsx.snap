// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage should have a consistent snapshot 1`] = `
<div>
  <main
    class="pages__Wrapper-sc-9oj6z6-0 iFtVcy"
  >
    <svg
      class="pages__Background-sc-9oj6z6-1 pYaWV"
      height="400"
      viewBox="0 0 1024 1024"
      width="400"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M 512.00,359.18
 C 512.00,359.18 512.00,359.18 512.00,359.18
 596.40,359.18 664.81,427.60 664.82,512.00
 664.82,512.00 664.82,512.00 664.82,512.00
 664.82,596.40 596.40,664.81 512.00,664.82
 512.00,664.82 512.00,664.82 512.00,664.82
 427.60,664.82 359.19,596.40 359.18,512.00
 359.18,512.00 359.18,512.00 359.18,512.00
 359.18,427.60 427.60,359.18 512.00,359.18
 512.00,359.18 512.00,359.18 512.00,359.18
 512.00,359.18 512.00,359.18 512.00,359.18 Z
 M 512.00,410.12
 C 455.73,410.12 410.12,455.73 410.12,512.00
 410.12,568.27 455.73,613.88 512.00,613.88
 568.27,613.88 613.88,568.27 613.88,512.00
 613.88,455.73 568.27,410.12 512.00,410.12
 512.00,410.12 512.00,410.12 512.00,410.12 Z
 M 806.16,217.84
 C 884.17,295.86 928.00,401.67 928.00,512.00
 928.00,622.33 884.17,728.14 806.16,806.16
 728.14,884.17 622.33,928.00 512.00,928.00
 401.67,928.00 295.86,884.17 217.84,806.16
 139.83,728.14 96.00,622.33 96.00,512.00
 96.00,401.67 139.83,295.86 217.84,217.84
 295.86,139.83 401.67,96.00 512.00,96.00
 622.33,96.00 728.14,139.83 806.16,217.84 Z
 M 332.02,512.00
 C 332.02,512.00 138.45,512.00 138.45,512.00
 138.45,611.07 177.81,706.09 247.86,776.14
 317.91,846.19 412.93,885.55 512.00,885.55
 611.07,885.55 706.09,846.19 776.14,776.14
 846.19,706.09 885.55,611.07 885.55,512.00
 885.55,512.00 885.55,512.00 885.55,512.00
 885.55,512.00 691.98,512.00 691.98,512.00
 691.98,412.60 611.40,332.02 512.00,332.02
 412.60,332.02 332.02,412.60 332.02,512.00 Z"
        fill="#000000"
        id="pkmn-go-menu"
        stroke="none"
        stroke-width="1"
      />
    </svg>
    <div
      class="pages__Search-sc-9oj6z6-2 lpizNx"
      id="search-bar"
    >
      <div
        class="MuiFormControl-root MuiTextField-root"
        style="width: 100%;"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink="false"
        >
          Search Pokemon
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <main
      class="FilterButtons__Wrapper-sc-mrsqbp-1 cVlVGQ pages__Filters-sc-9oj6z6-3 kmOafn"
      id="filters"
    >
      <div
        class="PokeballButton__PokeballWrapper-sc-z0qx2i-0 jUctph FilterButtons__Pokeball-sc-mrsqbp-0 ivNgsA"
        data-t-is-captured="false"
        title="Not captured"
      >
        <svg
          class="PokeballButton__PokeballSVG-sc-z0qx2i-1 fWWEza"
          height="45"
          viewBox="25 25 50 50"
          width="45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M 30 50
 a 1 1 1 0 1 40 0
 h-12.5
 a 1 1 1 0 0 -15 0
 z"
            fill="#f00"
            stroke="#222"
          />
          <circle
            cx="50"
            cy="50"
            fill="#222"
            r="5"
            stroke="#222"
          />
          <path
            d="M 30 50
 a 1 1 1 0 0 40 0
 h-12.5
 a 1 1 1 0 1 -15 0
 z"
            fill="#fff"
            stroke="#222"
          />
        </svg>
      </div>
      <div
        class="PokeballButton__PokeballWrapper-sc-z0qx2i-0 jUctph FilterButtons__Pokeball-sc-mrsqbp-0 ivNgsA"
        data-t-is-captured="true"
        title="Captured"
      >
        <svg
          class="PokeballButton__PokeballSVG-sc-z0qx2i-1 iJvgOH"
          height="45"
          viewBox="25 25 50 50"
          width="45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M 30 50
 a 1 1 1 0 1 40 0
 h-12.5
 a 1 1 1 0 0 -15 0
 z"
            fill="#f00"
            stroke="#222"
          />
          <circle
            cx="50"
            cy="50"
            fill="#222"
            r="5"
            stroke="#222"
          />
          <path
            d="M 30 50
 a 1 1 1 0 0 40 0
 h-12.5
 a 1 1 1 0 1 -15 0
 z"
            fill="#fff"
            stroke="#222"
          />
        </svg>
      </div>
    </main>
    <main
      class="PokemonsList__Wrapper-sc-1vm6xrc-0 gEFRsL pages__List-sc-9oj6z6-4 hHZKqv"
      id="pokemon-list"
    >
      <div
        style="overflow: visible; height: 0px; width: 0px;"
      >
        <div
          style="position: relative; height: -40px; width: -40px; overflow: auto; will-change: transform; direction: ltr;"
        >
          <div
            style="height: 105px; width: 100%;"
          >
            <span
              class="PokemonsList__ListItem-sc-1vm6xrc-2 ksvtKZ"
              data-t-list-item="bulbasaur"
              style="position: absolute; left: 0px; top: 0px; height: 35px; width: 100%;"
            >
              Bulbasaur
            </span>
            <div
              class="PokeballButton__PokeballWrapper-sc-z0qx2i-0 jUctph PokemonsList__Pokeball-sc-1vm6xrc-1 jVDyeK"
              data-t-is-captured="false"
              data-t-pokeball="list-bulbasaur"
              style="position: absolute; right: 0px; top: 0px; height: 35px;"
            >
              <svg
                class="PokeballButton__PokeballSVG-sc-z0qx2i-1 fWWEza"
                height="30"
                viewBox="25 25 50 50"
                width="30"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M 30 50
 a 1 1 1 0 1 40 0
 h-12.5
 a 1 1 1 0 0 -15 0
 z"
                  fill="#f00"
                  stroke="#222"
                />
                <circle
                  cx="50"
                  cy="50"
                  fill="#222"
                  r="5"
                  stroke="#222"
                />
                <path
                  d="M 30 50
 a 1 1 1 0 0 40 0
 h-12.5
 a 1 1 1 0 1 -15 0
 z"
                  fill="#fff"
                  stroke="#222"
                />
              </svg>
            </div>
            <span
              class="PokemonsList__ListItem-sc-1vm6xrc-2 cDyZEM"
              data-t-list-item="ivysaur"
              style="position: absolute; left: 0px; top: 35px; height: 35px; width: 100%;"
            >
              Ivysaur
            </span>
            <div
              class="PokeballButton__PokeballWrapper-sc-z0qx2i-0 jUctph PokemonsList__Pokeball-sc-1vm6xrc-1 jVDyeK"
              data-t-is-captured="false"
              data-t-pokeball="list-ivysaur"
              style="position: absolute; right: 0px; top: 35px; height: 35px;"
            >
              <svg
                class="PokeballButton__PokeballSVG-sc-z0qx2i-1 fWWEza"
                height="30"
                viewBox="25 25 50 50"
                width="30"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M 30 50
 a 1 1 1 0 1 40 0
 h-12.5
 a 1 1 1 0 0 -15 0
 z"
                  fill="#f00"
                  stroke="#222"
                />
                <circle
                  cx="50"
                  cy="50"
                  fill="#222"
                  r="5"
                  stroke="#222"
                />
                <path
                  d="M 30 50
 a 1 1 1 0 0 40 0
 h-12.5
 a 1 1 1 0 1 -15 0
 z"
                  fill="#fff"
                  stroke="#222"
                />
              </svg>
            </div>
            <span
              class="PokemonsList__ListItem-sc-1vm6xrc-2 ksvtKZ"
              data-t-list-item="venusaur"
              style="position: absolute; left: 0px; top: 70px; height: 35px; width: 100%;"
            >
              Venusaur
            </span>
            <div
              class="PokeballButton__PokeballWrapper-sc-z0qx2i-0 jUctph PokemonsList__Pokeball-sc-1vm6xrc-1 jVDyeK"
              data-t-is-captured="false"
              data-t-pokeball="list-venusaur"
              style="position: absolute; right: 0px; top: 70px; height: 35px;"
            >
              <svg
                class="PokeballButton__PokeballSVG-sc-z0qx2i-1 fWWEza"
                height="30"
                viewBox="25 25 50 50"
                width="30"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M 30 50
 a 1 1 1 0 1 40 0
 h-12.5
 a 1 1 1 0 0 -15 0
 z"
                  fill="#f00"
                  stroke="#222"
                />
                <circle
                  cx="50"
                  cy="50"
                  fill="#222"
                  r="5"
                  stroke="#222"
                />
                <path
                  d="M 30 50
 a 1 1 1 0 0 40 0
 h-12.5
 a 1 1 1 0 1 -15 0
 z"
                  fill="#fff"
                  stroke="#222"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        class="resize-triggers"
      >
        <div
          class="expand-trigger"
        >
          <div
            style="width: 1px; height: 1px;"
          />
        </div>
        <div
          class="contract-trigger"
        />
      </div>
    </main>
  </main>
</div>
`;
